### PR TITLE
Support for synchronizing secrets from tenant to LB cluster

### LINKED ISSUE #23: Propagation of TLS certificates from consumer to load-balancing cluster
**User Story**
As a KubeLB consumer, I'd like to secure my HTTP endpoints with TLS. For this purpose, I need to propagate TLS certificates for my routes from the consumer cluster to the load-balancing cluster.

**Acceptance criteria**

- [x] Propagation of TLS certificates from consumer to load-balancing cluster. 
- [x] Documentation

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:

With this PR, we want to add the propagation feature of TLS certificates from the consumer to the load-balancing cluster. The design has been kept simple to avoid any over-engineering. Since, in the future, we want to lean towards sources such as vaults, external secrets, etc, managed by the administrators for their tenants.

In this feature, we are creating a new resource, `SyncSecret`, which is simply a wrapper over Kubernetes Secret. The custom resource helps us ensure that we are not exposing any secrets from the LB cluster to the tenants.

### Native Kubernetes Secret:
```
kind: Secret
apiVersion: v1
metadata:
  name: mongodb-credentials
stringData:
  mongodb-password: "123456"
  mongodb-root-password: "123456"
type: Opaque
```

### Converted to a Sync Secret:
```
kind: SyncSecret
apiVersion: kubelb.k8c.io/v1alpha1
metadata:
  name: mongodb-credentials
stringData:
  mongodb-password: "123456"
  mongodb-root-password: "123456"
type: Opaque
```

The SyncSecret is then propagated to the tenant namespace in LB cluster for consumption. 

## Automation
This process can also be automated by enabling  the flag`enable-secret-synchronizer` for the CCM. This will enable a controller that automatically converts secrets labeled with `kubelb.k8c.io/managed-by: kubelb` to SyncSecrets in the tenant cluster.

**Which issue(s) this PR fixes**:

Fixes #23

**What type of PR is this?**

/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
* `SyncSecret` resource has been introduced that can be used to synchronize a secret from tenant to LB cluster
* Secret synchronizer controller can be enabled by the flag `enable-secret-synchronizer` for CCM. Enable automatically converting Secrets labeled with `kubelb.k8c.io/managed-by: kubelb` to SyncSecrets.  This controller requires elevated access to secrets in the tenant cluster to perform CRUD operations.
```

**Documentation**:

```documentation
TBD
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,6 +72,7 @@ manifests: generate controller-gen ## Generate WebhookConfiguration, ClusterRole
 	$(CONTROLLER_GEN) rbac:roleName=kubelb-ccm paths="./internal/controllers/ccm/..." output:artifacts:config=config/ccm/rbac
 	$(CONTROLLER_GEN) rbac:roleName=kubelb paths="./internal/controllers/kubelb/..." output:artifacts:config=config/kubelb/rbac
 	$(CONTROLLER_GEN) crd webhook paths="./..." output:crd:artifacts:config=charts/kubelb-manager/crds
+	cp charts/kubelb-manager/crds/kubelb.k8c.io_syncsecrets.yaml charts/kubelb-ccm/crds/kubelb.k8c.io_syncsecrets.yaml
```
Comment (ahmedwaleedmalik) [inline Makefile:75]: Note for the reviewer:

I wanted to do this via Kustomize, but it was more complicated than I had thought. Settled for this one-liner instead.

File: charts/kubelb-ccm/values.yaml [hunk 2]
```diff
@@ -16,6 +16,8 @@ kubelb:
   # -- Enable the leader election.
   enableLeaderElection: true
   nodeAddressType: InternalIP
+  # -- Enable to automatically convert Secrets labelled with `kubelb.k8c.io/managed-by: kubelb` to Sync Secrets. This is used to sync secrets from tenants to the LB cluster in a controlled and secure way.
+  enableSecretSynchronizer: false
```
Comment (ahmedwaleedmalik) [inline charts/kubelb-ccm/values.yaml:20]: Note for the reviewer:

Disabled by default since it required CRUD access for secrets.

**Review by cnvergence — APPROVED** [review]
LGTM, I like the simplicity of this

**Pull request merged**